pkg/build/build.go

```diff
@@ -184,29 +255,46 @@ func (spkg Subpackage) PackageURL(distro, packageVersionWithRelease string) stri
 }
 
 type SBOM struct {
+	// Optional: The language of the generated SBOM
 	Language string `yaml:"language"`
 }
 
 type Input struct {
+	// Optional: The human readable description of the input
 	Description string
-	Default     string
-	Required    bool
+	// Optional: The default value of the input. Required when the input is.
+	Default string
+	// Optional: A toggle denoting whether the input is required or not
+	Required bool
 }
 
+// The root melange configuration
 type Configuration struct {
-	Package     Package
+	// Package metadata
+	Package Package `yaml:"package"`
+	// The specification for the packages build environment
 	Environment apko_types.ImageConfiguration
-	Pipeline    []Pipeline   `yaml:"pipeline,omitempty"`
+	// Required: The list of pipelines that produce the package.
+	Pipeline []Pipeline `yaml:"pipeline,omitempty"`
+	// Optional: The list of subpackages that this package also produces.
 	Subpackages []Subpackage `yaml:"subpackages,omitempty"`
-	Data        []RangeData  `yaml:"data,omitempty"`
-	Secfixes    Secfixes     `yaml:"secfixes,omitempty"`
-	Advisories  Advisories   `yaml:"advisories,omitempty"`
-	Update      Update       `yaml:"update,omitempty"`
-
+	// Optional: An arbitrary list of data that can be used via templating in the
+	// pipeline
+	Data []RangeData `yaml:"data,omitempty"`
+	// Optional: The map of security fixes for this package keyed by the version
+	// found
+	Secfixes Secfixes `yaml:"secfixes,omitempty"`
+	// Optional: The map of advisories for this package
+	Advisories Advisories `yaml:"advisories,omitempty"`
```

Comment:
No change requested, just mentioning in case it impacts your plans at all: we'll be these fields soon, along with the related secfixes/advisory types below.